### PR TITLE
Enable VS Code Web Extension support

### PR DESCRIPTION
**Pull request by jpogran**
Fixes https://github.com/hashicorp/vscode-hcl/issues/111

This is primarily a syntax extension, so it should work in VS Code web ennvironments like github.dev and vscode.dev. Since we have actual code we show up as a normal extension and not one that can work on the web.

This commit adds a web extension file, supporting metadata to the package manifest, and ports bundling the extension to webpack. The result is a single VSIX that supports both desktop and web deployments.

In addition, this also updates all the VS Code task and launch configurations so that the extension can be tested locally both as a desktop or web extension.

**Note** that there is now a `src/extension.ts` and a `src/web/extension.ts` that appear identical. The two locations are required along with the changes inside package.json in order for the extension to be bundled properly for the two different environments. Most importantly, they allow webpack to properly include either the browser or desktop version of @vscode/extension-telemetry in the approprite extension. Consult https://code.visualstudio.com/api/extension-guides/web-extensions#web-extension-anatomy for more information about these requirements.

To test follow:

- Local web: Run `npm run run-in-browser`
- vscode.dev: Follow https://code.visualstudio.com/api/extension-guides/web-extensions#test-your-web-extension-in-on-vscode.dev
- Desktop: Launch 'Run Web Extension' Debug configuration

Local Web Install:

image

Developer Install on vscode.dev:

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "displayName": "HashiCorp HCL",
   "description": "HashiCorp HCL syntax",
   "appInsightsKey": "885372d2-6f3c-499f-9d25-b8b219983a52",
-  "version": "0.1.1",
+  "version": "0.1.0",
```
Comment (radeksimko) [inline package.json:8]: 👀

File: package.json [hunk 2]
```diff
@@ -66,43 +67,46 @@
...
     "prettier": "^2.5.1",
+    "process": "^0.11.10",
+    "ts-loader": "^9.3.0",
     "ts-node": "^10.6.0",
-    "typescript": "^4.5.4",
-    "vsce": "^2.6.7"
+    "typescript": "^4.7.3",
+    "vsce": "^2.9.1",
+    "webpack": "^5.72.1",
+    "webpack-cli": "^4.9.2"
```
Comment (radeksimko) [inline package.json:110]: Were all these dependency updates necessary to enable web support? If not - would you mind doing these updates in a separate PR?

File: package.json (hunk 1 above)
Comment (jpogran) [reply package.json:8]: Opps. Artifact of having to increment to test installation

**Comment by jpogran** [comment]
> That makes sense in principle, but could we not have `src/common` with shared logic, which both files would import?
> 

In general, yes, but the major concern is the telemetry import which we need to point to the browser implementation of telemetry for our web extension and the node implementation for our desktop extension. It was difficult enough getting it all to bundle correctly in either scenario that I didn't attempt DRY here.

I'll spend some more time and see.

> It appears this is roughly how the telemetry package is structured: https://github.com/microsoft/vscode-extension-telemetry/tree/main/src/

This is exactly the part that allows us to still use telemetry in the web version.

**Comment by jpogran** [comment]
@radeksimko the extraction of common logic worked!

File: package.json [hunk 3]
```diff
@@ -44,6 +44,7 @@
     "workspaceContains:**/*.hcl"
   ],
   "main": "./out/extension",
+  "browser": "./out/web/extension",
```
Comment (radeksimko) [inline package.json:47]: Nitpick: Have you considered making it either

```json
  "main": "./out/desktop/extension",
  "browser": "./out/web/extension",
```
or
```json
  "main": "./out/desktop",
  "browser": "./out/web",
```
just for readability/clarity to highlight the fact the logic in the other file is for desktop?

**Review by radeksimko — APPROVED** [review]
Just one question/suggestion, but not blocking.

File: package.json (hunk 3 above)
Comment (jpogran) [reply package.json:47]: I'd be OK with that except that the "desktop" also is used with Codespaces and the Remote extensions. Technically "node" is more accurate, but I think that sounds more confusing.

Comment (radeksimko) [reply package.json:47]: Hmmm, I see - that's a good point, and I agree `node` doesn't sound much clearer. Feel free to leave it as is! 😅

**Pull request merged**